src/SPC/builder/unix/library/bzip2.php

```diff
@@ -8,9 +8,14 @@ trait bzip2
 {
     protected function build(): void
     {
+        $cc = getenv('CC');
+        $ar = getenv('AR');
+        if ($ar === false) {
+            $ar = 'ar';
+        }
         shell()->cd($this->source_dir)
             ->exec("make PREFIX='" . BUILD_ROOT_PATH . "' clean")
-            ->exec("make -j{$this->builder->concurrency} PREFIX='" . BUILD_ROOT_PATH . "' libbz2.a")
+            ->exec("make -j{$this->builder->concurrency} PREFIX='" . BUILD_ROOT_PATH . "' CC={$cc} AR={$ar} libbz2.a")
```

Comment:
does this actually change anything? it should use the environment variables anyway

Reply:
Yes, it's a bit strange. It should have used env instead. But if I don't specify CC, the compilation result will have a linking error at the end, indicating `undefined reference fopen64`. When I looked for the corresponding source code in the bz code, I found that line of code is fopen().

Reply:
Yeah the ancient bzip2 1.0x source (which we use) defines the variables without checking the environment, so we need to pass them with the command line.

I will create a helper method to build the correct string for this.